### PR TITLE
V3.0.0 updates

### PR DESCRIPTION
**Pull request by paliwalparitosh**
- Restructure the terraform code
- Helm resource updates as per changes in helm-chart in v3
- Metrics collection via management agent
- Marketplace app UI updates
- Support for OCI livelab
- Multiple Other Minor updates

### CODE REVIEW

File: terraform/modules/helm/helm.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  helm_inputs = {
+    # global
+    "global.namespace"           = var.deploy_mushop_config ? "livelab-test" : var.kubernetes_namespace
+    "global.kubernetesClusterID" = var.oke_cluster_ocid
+
+    # oci-onm-logan
+    "oci-onm-logan.ociLANamespace"        = var.oci_la_namespace
+    "oci-onm-logan.ociLALogGroupID"       = var.oci_la_logGroup_id
+    "oci-onm-logan.image.url"             = var.container_image_url
+    "oci-onm-logan.kubernetesClusterName" = local.oke_cluster_name
```
Comment (santhoshkvuda) [inline terraform/modules/helm/helm.tf:21]: Set this at global.

Comment (paliwalparitosh) [reply terraform/modules/helm/helm.tf]: **kubernetesClusterName** was set to global scope. 
Validation - Stack is successfully deploying post this change.

File: terraform/modules/helm/inputs.tf [hunk 2]
```diff
@@ -1,72 +1,103 @@
...
-  type    = string
-  default = "/var/log"
-}
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+####
+## Switches
+####
+
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/modules/helm/inputs.tf:7]: Comments included for all the switches. Additionally, moved all debug/dev switches to `debug-inputs.tf`

File: terraform/modules/helm/inputs.tf [hunk 3]
```diff
@@ -1,72 +1,103 @@
...
+variable "oci_la_namespace" {
+  type = string
+}
+
+####
+##  Fluentd Configuration
+####
+
+# OCI LA Fluentd Container Image
+variable "container_image_url" {
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/modules/helm/inputs.tf:61]: updated.

File: terraform/modules/helm/inputs.tf [hunk 4]
```diff
@@ -1,72 +1,103 @@
...
+}
+
+# Fluentd Base Directory
+variable "fluentd_baseDir_path" {
+  type    = string
+  default = "/var/log"
+}
+
+####
+##  MACS Configuration
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/modules/helm/inputs.tf:73]: updated

File: terraform/modules/iam/iam.tf [hunk 5]
```diff
@@ -11,17 +11,18 @@ locals {
...
   # Policy
-  uuid_policy       = md5("${local.dynamic_group_name}${local.la_compartment_name}")
-  policy_name       = "oci-kubernetes-monitoring-${local.uuid_policy}"
-  policy_desc       = "Auto generated by Resource Manager Stack - oci-kubernetes-monitoring. Allows OKE Dynamic Group - ${local.dynamic_group_name} to upload data to Logging Analytics Service in ${local.la_compartment_name} compartment."
-  policy_scope      = var.root_compartment_ocid == var.oci_la_logGroup_compartment_ocid ? "tenancy" : "compartment ${local.la_compartment_name}"
-  policy_statements = ["Allow dynamic-group ${local.dynamic_group_name} to {LOG_ANALYTICS_LOG_GROUP_UPLOAD_LOGS} in ${local.policy_scope}"]
-
+  uuid_policy          = md5("${local.dynamic_group_name}${local.la_compartment_name}")
+  policy_name          = "oci-kubernetes-monitoring-${local.uuid_policy}"
+  policy_desc          = "Auto generated by Resource Manager Stack - oci-kubernetes-monitoring. Allows OKE Dynamic Group - ${local.dynamic_group_name} to upload data to Logging Analytics Service in ${local.la_compartment_name} compartment."
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/modules/iam/iam.tf:21]: updated the policy description.

File: terraform/modules/iam/iam.tf [hunk 6]
```diff
@@ -11,17 +11,18 @@ locals {
...
+  policy_scope         = var.root_compartment_ocid == var.oci_la_logGroup_compartment_ocid ? "tenancy" : "compartment ${local.la_compartment_name}"
+  macs_agent_policy    = ["Allow dynamic-group ${local.dynamic_group_name} to use METRICS in ${local.policy_scope} WHERE target.metrics.namespace = 'mgmtagent_kubernetes_metrics'"]
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/modules/iam/iam.tf:23]: updated across the terraform modules.

File: terraform/oke/main.tf [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    oci = oci.home_region
+  }
+}
+
+module "management_agent" {
+  source           = "./modules/macs"
+  uniquifier       = md5(var.oke_cluster_ocid)
+  compartment_ocid = var.oci_la_compartment_ocid
+
+  count = var.enable_macs ? 1 : 0
```
[the remark this answers is not in the document]
Comment (paliwalparitosh) [reply terraform/oke/main.tf:42]: - remove `enable_macs`  input
- Replaced with helm release module control switch/variable as it's useful to disable it while testing.

**Review by santhoshkvuda — APPROVED** [review]

**Review by palukuri — APPROVED** [review]

**Pull request merged**